modules/layouts/LayoutCS.chpl

```diff
@@ -328,13 +330,15 @@ class CSDom: BaseSparseDomImpl {
 
   proc dsiFirst {
     if _nnz == 0 then return (parentDom.low) - (1,1);
-    const _low = nnzDom.low;
+    const _low_idx = nnzDom.low;
+    var _low = idx[_low_idx];
+    // find the corresponding index into startIdx
     for i in startIdxDom {
-      if startIdx[i] > _low {
+      if startIdx[i] > _low_idx {
         if this.compressRows then
-          return (i-1, idx[idx.domain.low]);
+          return (i-1, _low);
         else
-          return (idx[idx.domain.low], i-1);
+          return (_low, i-1);
```

Comment:
Just so I understand, the change here is changing the behavior of `.first`, right? (And I remember seeing some discussion about that, was that in an issue?)

Reply:
This isn't changing the behavior.  I just broke down the process of getting the low_index and then the _low coordinate.  I am seeing now that the explanation is a bit confusing so I am going to put it back to what it was before.

https://github.com/chapel-lang/chapel/issues/17327 and https://github.com/chapel-lang/chapel/issues/17333 are the issues that were created to summarize the discussion about first and last in sparse and dense domains respectively.